param/parameterized.py

```diff
@@ -894,6 +897,23 @@ def _call_watcher(self_, watcher, change):
             watcher.fn(**{change.name: change.new})
 
 
+    def trigger(self_, *param_names):
+        """
+        Trigger watchers for the given set of parameter names. Watchers
+        will be triggered whether or not the parameter values have
+        actually changed.
+        """
+        changes = self_.cls.param._changes
+        watchers = self_.cls.param._watchers
+        self_.cls.param._changes  = []
+        self_.cls.param._watchers = []
+        param_values = self_.get_param_values()
+        params = {name:value for name, value in param_values if name in param_names}
```

Comment:
Currently this will silently ignore non-existent parameters, either you need to ensure they all exist or simply use:

```python
param_values = dict(self_.get_param_values())
params = {name: param_values[name] for name in param_names}
```